pkg/devfile/parser/configurables.go

```diff
@@ -60,17 +60,19 @@ func (d DevfileObj) RemoveEnvVars(keys []string) (err error) {
 	return d.WriteYamlDevfile()
 }
 
-// SetPorts converts ports to endpoints, adds to a devfile
-func (d DevfileObj) SetPorts(ports ...string) error {
+// SetPorts accepts a map of container name and the port numbers to be set;
+// it converts ports to endpoints, and adds to a devfile
```

Comment:
please update the description as this PR changes the existing behavior. currently it will set the `ports` to all container components, and this change will look for specific component and add the endpoint.